Sources/SwiftSyntaxMacros/MacroSystem.swift

```diff
@@ -438,6 +439,23 @@ extension MacroApplication {
       }
     }
 
+    let extensionMacroAttrs = getMacroAttributes(attachedTo: decl.as(DeclSyntax.self)!, ofType: ExtensionMacro.Type.self)
+    let extendedTypeSyntax = TypeSyntax("\(extendedType.trimmed)")
+    for (attribute, extensionMacro) in extensionMacroAttrs {
+      do {
+        let newExtensions = try extensionMacro.expansion(
+          of: attribute,
+          attachedTo: decl,
+          providingExtensionsOf: extendedTypeSyntax,
+          in: context
+        )
+
+        extensions.append(contentsOf: newExtensions.map(DeclSyntax.init))
+      } catch {
+        context.addDiagnostics(from: error, node: attribute)
+      }
+    }
+
```

Comment:
I'm not entirely sure what the assumptions are about how we're structuring code in this project, but I wonder whether it might be better to extract this code into a separate method. The name and doc-comment of this method suggest that the responsibility of this method is solely to expand conformances - I guess expanding extensions could be seen as a bit different responsibility. Just a thought! 💜

Reply:
Conformance macros are effectively a special case of extension macros. I'm going to take Doug's suggestion [here](https://github.com/apple/swift-syntax/pull/1859#discussion_r1245570504) in a follow-up PR and make `ConformanceMacro` refine `ExtensionMacro` so that they can be implemented in terms of extension macros. As part of that, I will clean up this code to only operate on `ExtensionMacro`, since `ConformanceMacro` will be a refinement of it!